launcher/src/test/java/org/apache/brooklyn/launcher/CleanOrphanedLocationsIntegrationTest.java

```diff
@@ -66,13 +72,26 @@
     private Set<ManagementContext> mgmts;
     private ManagementContext mgmt;
 
+    private String copyResourcePathToTempPath(String resourcePath) {
+        BundleMaker bm = new BundleMaker(ResourceUtils.create(this));
+        bm.setDefaultClassForLoading(CleanOrphanedLocationsIntegrationTest.class);
+        File jar = bm.createJarFromClasspathDir(resourcePath);
+        File output = Os.newTempDir("brooklyn-test-resouce-from-"+resourcePath);
+        try {
+            ArchiveUtils.extractZip(new ZipFile(jar), output.getAbsolutePath());
+            return output.getAbsolutePath();
+        } catch (Exception e) {
+            throw Exceptions.propagate(e);
+        }
+    }
+    
     @BeforeMethod(alwaysRun = true)
     @Override
     public void setUp() throws Exception {
         super.setUp();
-        persistenceDirWithOrphanedLocations = getClass().getResource(PERSISTED_STATE_PATH_WITH_ORPHANED_LOCATIONS).getFile();
-        persistenceDirWithoutOrphanedLocations = getClass().getResource(PERSISTED_STATE_PATH_WITHOUT_ORPHANED_LOCATIONS).getFile();
-        persistenceDirWithMultipleLocationsOccurrence = getClass().getResource(PERSISTED_STATE_PATH_WITH_MULTIPLE_LOCATIONS_OCCURRENCE).getFile();
+        persistenceDirWithOrphanedLocations = copyResourcePathToTempPath(PERSISTED_STATE_PATH_WITH_ORPHANED_LOCATIONS);
+        persistenceDirWithoutOrphanedLocations = copyResourcePathToTempPath(PERSISTED_STATE_PATH_WITHOUT_ORPHANED_LOCATIONS);
+        persistenceDirWithMultipleLocationsOccurrence = copyResourcePathToTempPath(PERSISTED_STATE_PATH_WITH_MULTIPLE_LOCATIONS_OCCURRENCE);
```

Comment:
There's the [File(URI)](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#File(java.net.URI)) constructor. That's tests so it's pretty safe to assume they execute from the file system.

Since what you wrote works correctly I'm happy to leave it as is.